client/concordclient/src/concord_client.cpp

```diff
@@ -147,9 +147,25 @@ void ConcordClient::subscribe(const SubscribeRequest& sub_req,
   if (std::holds_alternative<EventGroupRequest>(sub_req.request)) {
     ::client::thin_replica_client::SubscribeRequest trc_request;
     trc_request.event_group_id = std::get<EventGroupRequest>(sub_req.request).event_group_id;
-    trc_->Subscribe(trc_request);
+    try {
+      trc_->Subscribe(trc_request);
+    } catch (UpdateNotFound& e) {
+      throw;
+    } catch (OutOfRangeSubscriptionRequest& e) {
+      throw;
+    } catch (InternalError& e) {
+      throw;
+    }
```

Comment:
Given that you don't do anything we the exceptions, you can simply get rid of the try-catch block.
Same below.

Reply:
We throw the same exception, so that it can be caught by the event_service.cpp.

Reply:
The exceptions are thrown already, you don't have to catch and throw them again.